contributed_definitions/NXapm.nxdl.xml

```diff
@@ -0,0 +1,1129 @@
+                             Apart from the classical approach to use AMETEK software for this
+                             processing step, a number of open-source alternative tools exists.
+                        </doc>
+                        <attribute name="version" type="NX_CHAR">
+                            <doc>
+                                 Program version plus build number, commit hash, or description of
+                                 an ever persistent resource where the source code of the program
+                                 and build instructions can be found so that the program can be
+                                 configured in such a manner that the result file is ideally
+                                 recreatable yielding the same results.
+                            </doc>
+                        </attribute>
+                    </field>
+                    <group type="NXdata">
+                        <doc>
+                             A default three-dimensional histogram of the total number of ions in each bin.
+                        </doc>
+                    </group>
+                </group>
+            </group>
+            <group name="ranging" recommended="true" type="NXprocess">
+                <doc>
+                     Data post-processing step in which elemental, isotopic,
+                     and/or molecular identities are assigned to the ions.
+                     The documentation of these steps is based on ideas
+                     described in the literature:
+                     * [M. K. Miller](https://doi.org/10.1002/sia.1719)
+                     * [D. Haley et al.](https://doi.org/10.1017/S1431927620024290)
+                     * [M. Kühbach et al.](https://doi.org/10.1017/S1431927621012241)
+                </doc>
+                <field name="program" type="NX_CHAR">
+                    <doc>
+                         Given name of the program that was used to perform this computation.
+                         Apart from the classical approach to use AMETEK software for this
+                         processing step, a number of open-source alternative tools exists.
+                    </doc>
+                    <attribute name="version" type="NX_CHAR">
+                        <doc>
+                             Program version plus build number, commit hash, or description of
+                             an ever persistent resource where the source code of the program
+                             and build instructions can be found so that the program can be
+                             configured in such a manner that the result file is ideally
+                             recreatable yielding the same results.
+                        </doc>
+                    </attribute>
+                </field>
+                <field name="number_of_iontypes" type="NX_UINT" units="NX_UNITLESS">
```

Comment:
? `number_of_ion_types`  (just to be consistent)

Reply:
fixed